alt_e2eshark/onnx_tests/helper_classes.py

```
@@ -30,25 +30,21 @@
     get_sample_inputs_for_onnx_model,
 )
 
-# Checking if CACHE_DIR is set here will allow us to redefine
-# HF_HOME and HUGGINGFACE_HUB_CACHE without requiring them to
-# be set at shell level.
-parent_cache_dir = os.getenv("CACHE_DIR")
-
-os.environ["HF_HOME"] = "" if parent_cache_dir is None else parent_cache_dir
-os.environ["HUGGINGFACE_HUB_CACHE"] = (
-    "" if parent_cache_dir is None else parent_cache_dir
-)
-
-
 """This file contains several helpful child classes of OnnxModelInfo."""
 
 
 class HfDownloadableModel(OnnxModelInfo):
     """This class should be used to download models from Huggingface model hub."""
 
     def __init__(self, full_model_path, task_name, name, onnx_model_path):
-
+        # Checking if CACHE_DIR is set here will allow us to redefine
+        # HF_HOME and HUGGINGFACE_HUB_CACHE without requiring them to
+        # be set at shell level.
+        parent_cache_dir = os.getenv("CACHE_DIR")
+        os.environ["HF_HOME"] = "" if parent_cache_dir is None else parent_cache_dir
```

I think we should raise an exception here if CACHE_DIR is not set before setting the env vars, just like for other classes. This is subtle :) If I do not set CACHE_DIR before running an HF model, the HF_HOME variable will still be set (to "", interpreted as PWD by bash), and then the exception will be raised.

Having the check before changing the variables would save a great deal of unpleasantness, and cryptic log messages informing that the cache has been moved.

---

Yeah, that makes sense. It looks like we were also setting `parent_cache_dir` twice.